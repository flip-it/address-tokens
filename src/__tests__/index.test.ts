import { tokens, template } from "../index";

const input = {
  attention: "{firstName} {lastName}",
  road: "{addressLine1}\n{addressLine2}",
  city: "{city}",
  postcode: "{postalCode}",
  county: "{region}",
  state: "{region}",
  country: "{country}",
};

describe("tokens array", () => {
  test("SE", () => {
    expect(tokens(input, "SE")).toMatchSnapshot();
  });

  test("US", () => {
    expect(tokens(input, "US")).toMatchSnapshot();
  });

  test("AX", () => {
    expect(tokens(input, "AX")).toMatchSnapshot();
  });

  test("no input", () => {
    expect(tokens({}, "AX")).toMatchSnapshot();
  });

  test("invalid code", () => {
    expect(() => tokens(input, "X_")).toThrow();
  });

  test("empty code", () => {
    expect(() => tokens(input, "")).toThrow();
  });
});
const x = [
  ["firstName", "lastName"],
  ["addressLine1"],
  ["addressLine2"],
  ["city", "region", "postalCode"],
  ["country"],
];
describe("string template", () => {
  test("SE", () => {
    expect(template(input, "SE")).toMatchSnapshot();
  });

  test("US", () => {
    expect(template(input, "US")).toMatchSnapshot();
  });

  test("AX", () => {
    expect(template(input, "AX")).toMatchSnapshot();
  });

  test("no input", () => {
    expect(template({}, "AX")).toMatchSnapshot();
  });

  test("invalid code", () => {
    expect(() => template(input, "X_")).toThrow();
  });

  test("empty code", () => {
    expect(() => template(input, "")).toThrow();
  });
});
