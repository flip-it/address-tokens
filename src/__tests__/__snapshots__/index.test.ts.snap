// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string template AX 1`] = `
"{firstName} {lastName}
{addressLine1}
{addressLine2}
{postalCode} {city}
{country}"
`;

exports[`string template SE 1`] = `
"{firstName} {lastName}
{addressLine1}
{addressLine2}
{postalCode} {city}
{country}"
`;

exports[`string template US 1`] = `
"{firstName} {lastName}
{addressLine1}
{addressLine2}
{city}, {region} {postalCode}
{country}"
`;

exports[`string template no input 1`] = `""`;

exports[`tokens array AX 1`] = `
[
  [
    "firstName",
    "lastName",
  ],
  [
    "addressLine1",
  ],
  [
    "addressLine2",
  ],
  [
    "postalCode",
    "city",
  ],
  [
    "country",
  ],
]
`;

exports[`tokens array SE 1`] = `
[
  [
    "firstName",
    "lastName",
  ],
  [
    "addressLine1",
  ],
  [
    "addressLine2",
  ],
  [
    "postalCode",
    "city",
  ],
  [
    "country",
  ],
]
`;

exports[`tokens array US 1`] = `
[
  [
    "firstName",
    "lastName",
  ],
  [
    "addressLine1",
  ],
  [
    "addressLine2",
  ],
  [
    "city",
    "region",
    "postalCode",
  ],
  [
    "country",
  ],
]
`;

exports[`tokens array no input 1`] = `[]`;
